Automattic/jetpack
# Add new i18n fonts and rename them via filter

**Adding new fonts for better global support.**

An alternative to https://github.com/Automattic/jetpack/pull/27751

Related to adding & deprecating other fonts: https://github.com/Automattic/jetpack/pull/27441

See design discussion p9Jlb4-5Dj-p2 

![image](https://user-images.githubusercontent.com/87168/205674792-15759d42-db5e-48d1-9edd-a6c95acb5120.png)

#### Changes proposed in this Pull Request:
* Adds a way to filter Global Styles to provide different name from "font family". Google fonts expects Noto Sans fonts to be requested with font family such as "Noto Serif TC" instead of "Noto Serif (Traditional Chinese)". `wp_register_webfonts()` [does not support passing the name](https://github.com/WordPress/gutenberg/issues/46398), although [Global Style Schema](https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-living/#typography) does support the name and it thus works in the UI once provided via filter. Once the support for "name" is added, we can go via simpler approach implemented in https://github.com/Automattic/jetpack/pull/27751 instead.
* Moves font list to a getter function to be able to:
   * Translate new font names
   * Filter the list for both functions where the list is now used
* Adds new fonts at the end of the fonts list:

    ```
	'Alexandria'           => 'Alexandria (Arabic)',
	'IBM Plex Sans Arabic' => 'IBM Plex Sans (Arabic)',
	'Noto Sans Hebrew'     => 'Noto Sans (Hebrew)',
	'Noto Sans HK'         => 'Noto Sans (Hong Kong)',
	'Noto Sans JP'         => 'Noto Sans (Japanese)',
	'Noto Sans KR'         => 'Noto Sans (Korean)',
	'Noto Sans SC'         => 'Noto Sans (Simplified Chinese)',
	'Noto Sans TC'         => 'Noto Sans (Traditional Chinese)',
	'Noto Sans Telugu'     => 'Noto Sans (Telugu)',
	'Noto Serif Hebrew'    => 'Noto Serif (Hebrew)',
	'Noto Serif HK'        => 'Noto Serif (Hong Kong)',
	'Noto Serif JP'        => 'Noto Serif (Japanese)',
	'Noto Serif KR'        => 'Noto Serif (Korean)',
	'Noto Serif SC'        => 'Noto Serif (Simplified Chinese)',
	'Noto Serif TC'        => 'Noto Serif (Traditional Chinese)',
    ```

Mukta, which is present in the screenshot above, is missing from the list since it wasn't available at Google Fonts.

#### Other information:


#### Jetpack product discussion



p9Jlb4-5Dj-p2

#### Does this pull request change what data or activity we track or use?



Nope.

#### Testing instructions:






* Test using non-blockbase theme (including child themes) because blockbase [disables Jetpack fonts](https://github.com/Automattic/themes/blob/36a128332420acf5501789cd892a068e0e12cf51/blockbase/inc/fonts/custom-fonts.php#L208-L216). Alternatively test using the [upcoming change](https://github.com/Automattic/themes/pull/6777) in blockbase.
* Ensure "Google fonts" module is enabled when testing with standalone Jetpack. At .com it is enabled already.
* Go to Appearance → Editor → Press "edit" on newer Gutenberg.
* Navigate to Styles panel and change fonts for the text or headings. See all the fonts in the dropdown:

    <img width="295" alt="image" src="https://user-images.githubusercontent.com/87168/206702864-82ba4375-9cc0-4277-bf8f-176c87d15821.png">

    <img width="587" alt="Screenshot 2022-12-09 at 14 07 08" src="https://user-images.githubusercontent.com/87168/206702587-73be05fa-2618-4d60-9e4b-9e4410871f17.png">

* Have some Japanese, Arabic, Chinese etc texts to test with and observe how specific fonts support these scripts beautifully.

    <img width="1371" alt="Screenshot 2022-12-09 at 14 08 03" src="https://user-images.githubusercontent.com/87168/206702507-5ceea2b9-9e4e-4311-a4cb-37f27d8140ea.png">

    <img width="1371" alt="Screenshot 2022-12-09 at 14 08 09" src="https://user-images.githubusercontent.com/87168/206702503-2aade2e9-1aa8-4d92-b0fe-42b6e9ed788f.png">

* Note also difference between Noto Sans and Noto Serif in the language you're testing, here's Japanese:

    <img width="997" alt="Screenshot 2022-12-15 at 13 44 05" src="https://user-images.githubusercontent.com/87168/207851131-3323a915-4cca-42c7-aabc-09d3809ac543.png">
    <img width="985" alt="Screenshot 2022-12-15 at 13 44 14" src="https://user-images.githubusercontent.com/87168/207851144-52e30ad3-f4f2-406c-8cff-df389286e383.png">


* Ensure frontend of the site works, too.
* Observe how values are different from the actual names in the dropdown list:
    <img width="378" alt="Screenshot 2022-12-09 at 14 07 34" src="https://user-images.githubusercontent.com/87168/206702458-5b8e2200-b915-43e6-85ee-86035c1817d7.png">
